src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -117,6 +117,16 @@ public static class TaskExtensions {
         }
 
         /// <summary>
+        /// Waits for a task to complete within a given timeout. 
+        /// Returns task result or default(T) if task did not complete
+        /// within the timeout specified.
+        /// </summary>
+        public static T WaitTimeout<T>(this Task<T> task, int msTimeout) {
+            task.Wait(msTimeout);
+            return task.IsCompleted ? task.Result : default(T);
```

Comment:
Right. So make this return false if it timed out or true if it didn't, and then the caller can directly access `Result` in a way that unwraps exceptions (e.g. via `GetAwaiter().GetResult()`), or fall back to some default value that is not actually `default(T)`.

Alternatively, make it throw `TimeoutException` if it times out, and the caller has to catch/handle.